File: src/main/java/com/mingshashan/learn/lt/l009/Solution_009.java
package com.mingshashan.learn.lt.l009;

/**
 * Palindrome Number
 * Given an integer x, return true if x is palindrome integer.
 * <p>
 * An integer is a palindrome when it reads the same backward as forward.
 * <p>
 * For example, 121 is a palindrome while 123 is not.
 */
public class Solution_009 {


    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        // 特殊情况
        if (x < 10) {
            return true;
        }

        int temp = x;
        int t = 0;
        while (temp > 0) {
            t = t * 10 + (temp % 10);
            temp /= 10;
        }

        return t == x;
    }
}

File: src/main/java/com/mingshashan/learn/lt/l009/Solution_009.c

bool isPalindrome(int x){
    if (x < 0) {
        return false;
    }

    if (x < 10) {
        return true;
    }
    
    int temp = x;
    int p = 0;
    int max = 2147483647;

    while (temp) {
        if (p > ((max - temp % 10) / 10)) {
            return false;
        }
        p = p * 10 + temp % 10;
        temp /= 10;
    }

    return x == p;
}